src/utils/SharedUtils.js

```diff
@@ -0,0 +1,163 @@
+  xhr.onload = function() {
+    var status = xhr.status;
+    var response = xhr.response;
+  };
+  xhr.send(JSON.stringify(data));
+}
+
+export function updateUrl(propKey, propValue) {
+  var searchObject = parseUrlSearchString();
+
+  searchObject[propKey] = propValue;
+
+  var newSearchString = '?' + createUrlSearchString(searchObject);
+
+  history.pushState({}, "", window.location.origin + window.location.pathname + newSearchString);
+  localStorage[window.location.pathname.replace(/\//g,'')] = newSearchString;
+}
+
+export function parseFilterBarConfiguration(filterBarConfiguration, id) {
+  var parsedFilterBarConfiguration = {},
+      rawFilter,
+      rawFilters,
+      parsedFilters = {};
+
+  rawFilters = filterBarConfiguration.querySelector('dl.filters').querySelectorAll('dt.filter');
+
+  for (var i = 0; i < rawFilters.length; i++) {
+    rawFilter = rawFilters[i];
+    parsedFilters[rawFilter.getAttribute('data-uid')] = {
+      label: rawFilter.getAttribute('data-label'),
+      type: rawFilter.getAttribute('data-type'),
+      field: rawFilter.getAttribute('data-field'),
+      url: rawFilter.getAttribute('data-url'),
+      value: '',
+      enabled: false
+    };
+  }
+
+  parsedFilterBarConfiguration.id = id;
+  parsedFilterBarConfiguration.persistent = filterBarConfiguration.querySelector('dt.persistent').getAttribute('data-persistent');
+  parsedFilterBarConfiguration.searchUrl = filterBarConfiguration.querySelector('dt.search-url').getAttribute('data-url');
+  parsedFilterBarConfiguration.saveSearchUrl = filterBarConfiguration.querySelector('dt.save-search-url').getAttribute('data-url');
+  parsedFilterBarConfiguration.savedSearchUrl = filterBarConfiguration.querySelector('dt.saved-search-url').getAttribute('data-url');
+  parsedFilterBarConfiguration.exportResultsUrl = filterBarConfiguration.querySelector('dt.export-results-url').getAttribute('data-url');
+  parsedFilterBarConfiguration.filters = parsedFilters;
+
+  return parsedFilterBarConfiguration;
+}
+
+export function parseTableConfiguration(tableConfiguration, id) {
+  var parsedTableConfiguration = {},
+      rawColumns,
+      rawColumn,
+      parsedColumns = {};
+
+  rawColumns = tableConfiguration.querySelector('dl.columns').querySelectorAll('dt.column');
+
+  for (var i = 0; i < rawColumns.length; i++) {
+    rawColumn = rawColumns[i];
+    parsedColumns[rawColumn.getAttribute('data-field')] = {
+      heading: rawColumn.getAttribute('data-heading'),
+      type: rawColumn.getAttribute('data-type')
+    };
+  }
+
+  parsedTableConfiguration.id = id;
+  parsedTableConfiguration.dataUrl = tableConfiguration.querySelector('dt.data-url').getAttribute('data-url');
+  parsedTableConfiguration.columns = parsedColumns;
+
+  return parsedTableConfiguration;
+}
+
+export function parseUrlSearchString() {
+  // taken from https://github.com/sindresorhus/query-string/blob/master/query-string.js
+
+  var str = window.location.search;
```

Comment:
This is one of them. Since it was a couple of lines I just dropped it in until I can have a discussion about including external libraries in JS and then in Rails.